egui/src/widgets/text_edit.rs

```diff
@@ -138,6 +138,7 @@ impl CCursorPair {
 #[derive(Debug)]
 pub struct TextEdit<'t> {
     text: &'t mut String,
+    prev_text: String,
```

Comment:
I don't think this should be a member. `TextEdit` is just a builder pattern - you can declare `prev_text` in the `content_ui` function.